probes/webhooksWithoutTokenAuth/impl_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint:stylecheck
+package webhooksWithoutTokenAuth
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/clients"
+	"github.com/ossf/scorecard/v4/finding"
+)
+
+func Test_Run(t *testing.T) {
+	t.Parallel()
+	// nolint:govet
+	tests := []struct {
+		name     string
+		raw      *checker.RawResults
+		outcomes []finding.Outcome
+		err      error
+	}{
+		{
+			name: "Webhooks present with auth secret",
+			raw: &checker.RawResults{
+				WebhookResults: checker.WebhooksData{
+					Webhooks: []clients.Webhook{
+						{
+							Path:           "https://github.com/owner/repo/settings/hooks/1234",
+							ID:             1234,
+							UsesAuthSecret: true,
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomePositive,
+			},
+		},
+		{
+			name: "Webhooks present without auth secret",
+			raw: &checker.RawResults{
+				WebhookResults: checker.WebhooksData{
+					Webhooks: []clients.Webhook{
+						{
+							Path:           "https://github.com/owner/repo/settings/hooks/1234",
+							ID:             1234,
+							UsesAuthSecret: false,
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomeNegative,
+			},
+		},
+		{
+			name: "Multiple webhooks present, one without auth secret",
+			raw: &checker.RawResults{
+				WebhookResults: checker.WebhooksData{
+					Webhooks: []clients.Webhook{
+						{
+							Path:           "https://github.com/owner/repo/settings/hooks/1234",
+							ID:             1234,
+							UsesAuthSecret: false,
+						},
+						{
+							Path:           "https://github.com/owner/repo/settings/hooks/12345",
+							ID:             1234,
+							UsesAuthSecret: true,
+						},
+						{
+							Path:           "https://github.com/owner/repo/settings/hooks/12346",
+							ID:             1234,
+							UsesAuthSecret: true,
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomeNegative,
```

Comment:
we need 2 positive outcomes too, for the positive probes, right?

Reply:
No. The probe only returns a positive outcome if there are no webhooks without authentication.

Reply:
sorry, let me rephrase. If there are N webhooks, should we not get N findings, one for each webhook? If there's no hook, it would return NotApplicable. If there are hooks, you return one finding for each. The path of the hook can be used to populate the Location for the finding https://github.com/ossf/scorecard/blob/main/finding/finding.go#L95 (path and type https://github.com/ossf/scorecard/blob/main/finding/finding.go#L48-L49)

Reply:
This should no longer be relevant because of https://github.com/ossf/scorecard/pull/3522#issuecomment-1811514146